From c60cb3ccf979e45f3569edf6d3d804e00c75b185 Mon Sep 17 00:00:00 2001
From: Charlie Paul <Charlie.Paul@windriver.com>
Date: Thu, 4 May 2017 12:27:56 -0700
Subject: [PATCH 218/278] drivers/usb/dwc3: Initialize dma for axxia dwc3

The coherency bit needs to be set for USB ports on the
axxia boards to initialize. In 4.9 the initialization
of the DMA for USB changed and this patch properly
initializes the USB.

Signed-off-by: Charlie Paul <Charlie.Paul@windriver.com>
---
 drivers/usb/dwc3/dwc3-axxia.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/dwc3-axxia.c b/drivers/usb/dwc3/dwc3-axxia.c
index c2ac051..d47c46e 100644
--- a/drivers/usb/dwc3/dwc3-axxia.c
+++ b/drivers/usb/dwc3/dwc3-axxia.c
@@ -83,7 +83,7 @@ static int axxia_dwc3_remove(struct platform_device *pdev)
 arch_setup_pdev_archdata(struct platform_device *pdev)
 {
 	if (strncmp(pdev->name, "xhci-hcd", strlen("xhci-hcd")) == 0)
-		pdev->dev.archdata.dma_coherent = 1;
+		arch_setup_dma_ops(&pdev->dev, 0, 0, NULL, 1);
 }
 
 static const struct of_device_id adwc3_of_match[] = {
-- 
1.8.5.2.233.g932f7e4

